Libraries/LibLine/Editor.cpp

```diff
@@ -279,17 +282,23 @@ String Editor::get_line(const String& prompt)
                 m_refresh_needed = true;
                 continue;
             }
+
+            auto saved_errno = errno;
+
             perror("read failed");
-            // FIXME: exit()ing here is a bit off. Should communicate failure to caller somehow instead.
-            exit(2);
+            dbg() << "Read failed: " << strerror(saved_errno);
+
+            if (saved_errno == EINVAL || saved_errno == EBADF)
+                return Error::ReadFailure;
+
+            return Error::Unspecified;
```

Comment:
Why single out (double out?) those two error codes?

Reply:
because they're the only other error codes that `read()` can fail with in this scenario?

Reply:
To my (admittedly shit memory), read can fail with BADF, INTR, INVAL, FAULT and IO.
we're not doing any disk IO, so EIO shouldn't come out at all.
our buffer on the stack so FAULT is out too
INTR is handled, leaving the other two

Reply:
`read()` can fail for way more reasons than you can ever imagine. Any device driver or any magical fine can return whatever errors it can think of.

> we're not doing any disk IO

Aren't we?

Reply:
we're reading from `stdin`...
if the drivers spit out something weird, that's just an unspecified error...
I guess there's no need for `Unspecified` then, I'll drop that.

Reply:
> we're reading from `stdin`...

Aka whatever happens to be fd 0

Reply:
> Aka whatever happens to be fd 0

We're working on a level of abstraction here, LibLine assumes some stuff about that...I don't think worrying about things we don't consider legitimate read errors is that important.
Nonetheless, every error except INTR is just a read error now.